File: py110/small_problems_110_119/easy2/9.py
# https://launchschool.com/exercises/6051b11c

'''
Questions:


Input:
Output:
Explicit:
Implicit:


Pseudocode:

'''

def count_occurrences(vehicles):
    freq_dict = {}

    for vehicle in vehicles:
        freq_dict.setdefault(vehicle, 0)
        freq_dict[vehicle] += 1

    for vehicle, count in freq_dict.items():
        print(f"{vehicle} => {count}")


vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck',
            'motorcycle', 'motorcycle', 'car', 'truck']

count_occurrences(vehicles)

File: js210/simpleProblems/easy4/9.js
// https://launchschool.com/exercises/2b9f8b47

'use strict';

// function countOccurrences(listOfVehicles) {
//   let objMap = {};

//   for (let vehicle of listOfVehicles) {
//     objMap[vehicle] ? objMap[vehicle] += 1 : objMap[vehicle] = 1;
//   }

//   for (let [key, value] of Object.entries(objMap)) {
//     console.log(`${key} => ${value}`);
//   }
// }

function countOccurrences(listOfVehicles) {
  let vehicleMap = new Map();

  for (let vehicle of listOfVehicles) {
    vehicleMap.set(vehicle, (vehicleMap.get(vehicle) ?? 0) + 1);
  }

  vehicleMap.forEach((value, key) => console.log(`${key} => ${value}`));
}


const vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck',
                'motorcycle', 'suv', 'motorcycle', 'car', 'truck'];

countOccurrences(vehicles);

// console output
// car => 4
// truck => 3
// SUV => 1
// motorcycle => 2
// suv => 1